wicker/core/datasets.py

```diff
@@ -303,3 +313,248 @@ def dataset_size(self) -> int:
             int: total dataset size in bytes
         """
         return self._get_dataset_partition_size()
+
+
+class SFDataset(AbstractDataset):
+    """Loading dataset from Snowflake table"""
+
+    def __init__(
+        self,
+        dataset_name: str,
+        dataset_version: str,
+        dataset_partition_name: str,
+        table_name: str,
+        connection_parameters: Optional[Dict[str, str]] = None,
```

Comment:
from the consistent perspective, prob need to move this to the wicker config as the s3 dataset does